packages/aws-cdk-lib/aws-sqs/test/sqs.test.ts

```diff
@@ -753,6 +753,30 @@ describe('redriveAllowPolicy', () => {
     });
   });
 
+  test('explicit specification of redrive permission', () => {
+    const stack = new Stack();
+    new sqs.Queue(stack, 'Queue', {
+      redriveAllowPolicy: {
+        redrivePermission: sqs.RedrivePermission.DENY_ALL,
+      },
+    });
+
+    Template.fromStack(stack).templateMatches({
+      'Resources': {
+        'Queue4A7E3555': {
+          'Type': 'AWS::SQS::Queue',
+          'Properties': {
+            'RedriveAllowPolicy': {
+              'redrivePermission': 'denyAll',
+            },
+          },
+          'UpdateReplacePolicy': 'Delete',
+          'DeletionPolicy': 'Delete',
+        },
+      },
+    });
+  });
```

Comment:
Might be better with `test.each`.

```suggestion
  test.each([
    [sqs.RedrivePermission.ALLOW_ALL, 'allowAll'],
    [sqs.RedrivePermission.DENY_ALL, 'denyAll'],
  ])('redrive permission can be set to %s', (permission, expected) => {
    const stack = new Stack();
    new sqs.Queue(stack, 'Queue', {
      redriveAllowPolicy: {
        redrivePermission: permission,
      },
    });

    Template.fromStack(stack).templateMatches({
      'Resources': {
        'Queue4A7E3555': {
          'Type': 'AWS::SQS::Queue',
          'Properties': {
            'RedriveAllowPolicy': {
              'redrivePermission': expected,
            },
          },
          'UpdateReplacePolicy': 'Delete',
          'DeletionPolicy': 'Delete',
        },
      },
    });
  });
```